### ISSUE TITLE
Missing explicit compile dependency on Guava

### ISSUE THREAD

**Issue by baron1405** [commercehub-oss/gradle-avro-plugin#18]
When attempting to apply version 0.6.0 the build fails with:
```
Caused by: java.lang.NoClassDefFoundError: com/google/common/collect/ImmutableSet
	at com.commercehub.gradle.plugin.avro.GenerateAvroJavaTask.<clinit>(GenerateAvroJavaTask.java:28)
	at org.gradle.api.internal.DependencyInjectingInstantiator.newInstance(DependencyInjectingInstantiator.java:48)
	at org.gradle.api.internal.ClassGeneratorBackedInstantiator.newInstance(ClassGeneratorBackedInstantiator.java:36)
	at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:121)
```
This is a dependency on a Google Guava class but there is no explicit compile dependency on Guava. I suspect that the build of this project is succeeding by a transitive dependency on Guava. I believe that fix here is to add an explicit compile dependency on the desired version of Guava.

**Comment by davidmc24** [comment]
Thanks for the report.  What version of Gradle were you using?

**Comment by davidmc24** [comment]
Reproduced, working on a fix.

**Comment by baron1405** [comment]
Gradle 2.7

**Comment by davidmc24** [comment]
I really don't understand why Gradle is making Guava available at build-time for the plugin project but not runtime.  Regardless, I'll be pushing out a fix version shortly that doesn't use Guava, and will have a process in place to prevent this class of issue from easily recurring.

If you need to work-around with version 0.6.0, you can use the following `build.gradle` content to explicitly add a dependency on guava:

```groovy
buildscript {
    repositories {
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
    dependencies {
        classpath "com.google.guava:guava:18.0"
        classpath "com.commercehub.gradle.plugin:gradle-avro-plugin:0.6.0"
    }
}
apply plugin: "com.commercehub.gradle.plugin.avro"
```

**Comment by baron1405** [comment]
Thanks David for the quick turnaround!

**Comment by davidmc24** [comment]
Fixed in version 0.6.1, which should now be available.
